base/sort.jl

```diff
@@ -415,9 +415,9 @@ sort(v::AbstractVector; kws...) = sort!(copymutable(v); kws...)
 
 ## other iterables and fallback ##
 
-sort(s::String; kws...) = String(sort(collect(s); kws...))
+sort(s::String; kws...) = String(sort!(collect(s); kws...))
 sort(n::Number) = throw(MethodError(sort, n))
-sort(itr; kws...) = sort(collect(itr); kws...)
+sort(itr; kws...) = sort!(collect(itr); kws...)
```

Comment:
I figured `collect` always made a new object so it would be safe to modify that in place and avoid an extra copy. I didn't realize it was possible for `collect` to alias.